marlowe-playground-client/src/Blockly.js

```diff
@@ -83,8 +83,13 @@ exports.getBlockById_ = function (just, nothing, workspace, id) {
 }
 
 exports.workspaceXML_ = function (blockly, workspace) {
-    var dom = blockly.Xml.workspaceToDom(workspace);
-    return blockly.utils.xml.domToText(dom);
+    const isEmpty = workspace.getAllBlocks()[0].getChildren().length == 0;
+    if (isEmpty) {
```

Comment:
This feels a bit odd to me somehow. I think it's because the test isn't actually in the `if`-clause.

Would this be better?

```javascript
    const childCount = workspace.getAllBlocks()[0].getChildren().length;
    if (childCount == 0) {
```

...or even:

```javascript
    const children = workspace.getAllBlocks()[0].getChildren();
    if (children.length == 0) {
```

Reply:
Hmm, I'm not sure. This is a useful state to know, is the workspace currently empty (for the particular definition of this domain, i.e. it has a root block but nothing else). Then I do something based on whether or not it is empty, not based on how many children there are.